src/main/scala/where/Reference.scala

```diff
@@ -38,12 +39,22 @@ sealed trait Reference extends ToQueryWithIdentifiers {
 
 }
 
+sealed trait NodeOrRelationshipReference extends Reference
+
 object Reference {
-  implicit def toActionReference(reference: Reference): ActionReference =
-    ActionReference(reference)
+  implicit def toActionReference(reference: ObjectReference): ActionReference =
+    ActionNodeOrRelationship(reference)
+
+  implicit def toActionReference(reference: ReferenceWithProperty): ActionReference =
+    ActionNodeOrRelationship(reference)
+
+  implicit def objectReferencesToActionReferences(references: Seq[ObjectReference]): Seq[ActionReference] =
+    references map (ActionNodeOrRelationship.apply(_))
 
-  implicit def toActionReferenceSeq(references: Seq[Reference]): Seq[ActionReference] =
-    references map (ActionReference(_))
+  implicit def referencesWithPropertiesToActionReferences(
+    references: Seq[ReferenceWithProperty]
+  ): Seq[ActionReference] =
+    references map (ActionNodeOrRelationship.apply(_))
```

Comment:
yeah actually the reason `apply` even shows up there is because i was playing around with that and i forgot to remove the `apply` when switching back, i'll fix :/